### PR TITLE
fix: dassert if shared log is damaged when replica server init

### PR DESCRIPTION
**Pull request by Shuo-Jia**
### Releated issue:
https://github.com/XiaoMi/pegasus/issues/552

We found the shared log may be incomplete after one replica_server node down and restart,  for avoid some unexpected problem,  this pr `dassert` the case.  it can help us find problem in time when error happens.

### CODE REVIEW

File: src/dist/replication/lib/replica_stub.cpp [hunk 1]
```diff
@@ -598,30 +597,16 @@ void replica_stub::initialize(const replication_options &opts, bool clear /* = f
                smax);
 
         if (err == ERR_OK) {
-            if (smax != pmax) {
```
Comment (levy5307) [inline src/dist/replication/lib/replica_stub.cpp:601]: I guess smax is the max decree of share log, and pmax is the max decredd of private log. Is that right?

**Comment by levy5307** [comment]
So what caused this situation? Can we prevent this situation from happening, by changing some code?

File: src/dist/replication/lib/replica_stub.cpp (hunk 1 above)
Comment (Shuo-Jia) [reply src/dist/replication/lib/replica_stub.cpp:601]: Yeah

**Comment by Shuo-Jia** [comment]
> So what caused this situation? Can we prevent this situation from happening, by changing some code?

Now, I have find the casue

**Pull request merged**